File: Trees/rightSideView.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
};

vector<int> rightSideView(TreeNode *root)
{
    vector<int> ans;

    if (root == NULL)
    {
        return ans;
    }

    queue<TreeNode *> q;

    q.push(root);

    while (!q.empty())
    {
        int count = q.size();
        vector<int> curr;

        for (int i = 0; i < count; i++)
        {
            TreeNode *node = q.front();
            q.pop();

            curr.push_back(node->val);

            if (node->left != NULL)
            {
                q.push(node->left);
            }
            if (node->right != NULL)
            {
                q.push(node->right);
            }
        }

        ans.push_back(curr[curr.size() - 1]);
    }

    return ans;
}

File: Trees/Right_Side_View.py
from typing import Optional, List
from collections import deque


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result = []

        queue = deque()
        queue.append(root)

        while queue:
            rightSide = None

            for i in range(len(queue)):
                node = queue.popleft()

                if node:
                    rightSide = node
                    queue.append(node.left)
                    queue.append(node.right)
            if rightSide:
                result.append(rightSide.val)
        return result
